File: backend/src/WebApi/Models/UserModels.cs
namespace WebApi.Models;

public sealed record UserInfo(Guid Id, string Name, bool IsActive);

public sealed record UserDetailInfo(
    Guid Id,
    string FirstName,
    string LastName,
    string Name,
    bool IsActive,
    decimal MonthlyExpenseQuota);

public sealed record CreateUserRequest(
    string FirstName,
    string LastName,
    bool IsActive,
    decimal MonthlyExpenseQuota);

public sealed record UpdateUserRequest(
    string FirstName,
    string LastName,
    bool IsActive,
    decimal MonthlyExpenseQuota);

File: frontend/libs/data-access/src/lib/models/user.models.ts
export interface User {
  id: string;
  name: string;
  isActive: boolean;
}

export interface UserDetail {
  id: string;
  firstName: string;
  lastName: string;
  name: string;
  isActive: boolean;
  monthlyExpenseQuota: number;
}

export interface CreateUserRequest {
  firstName: string;
  lastName: string;
  isActive: boolean;
  monthlyExpenseQuota: number;
}

export interface UpdateUserRequest {
  firstName: string;
  lastName: string;
  isActive: boolean;
  monthlyExpenseQuota: number;
}

